markdown-link-check

```diff
@@ -102,8 +103,12 @@ function getInputs() {
         input.opts.verbose = (program.verbose === true);
         input.opts.retryOn429 = (program.retry === true);
         input.opts.aliveStatusCodes = program.alive;
-        // set the projectBaseUrl to the current working directory, so that `{{BASEURL}}` can be resolved to the project root.
-        input.opts.projectBaseUrl = `file://${process.cwd()}`;
+        if (program.projectBaseUrl) {
```

Comment:
Here, I'm trying to check if the argument was passed. I'm not sure if there's a canonical way to do this in JavaScript. `if(program.projectBaseUrl !== undefined)` seems like one option.

Reply:
If you want to explicitly check for undefined, here's some information undefined values in JavaScript: https://masteringjs.io/tutorials/fundamentals/undefined-check

I think what you have is fine, though. The if statement checks if the value is _truthy_, so it will cover both null and undefined values.

Reply:
On the other hand, I suppose it would fall to the false case if someone passed "false" or "0" as the base URL.

Reply:
Thanks for the pointers. I don't feel good about "false" or "0" making it through the simpler `if(program.projectBaseUrl)` check... there's something about the principle of least surprise there. It also seems reasonable to make the behavior as close to what we'd get in a strongly-typed language as possible. I think `!== undefined` is the move. We want to fall-through if the object doesn't have a `projectBaseUrl` property, or if it has that property and it's undefined, and that accomplishes that.

I considered the `in` operator, but I don't like how `projectBaseUrl in program` is `true` if `program.projectBaseUrl === undefined`.

Reply:
I misunderstood. `if(program.projectBaseUrl)` is the play. We don't want to treat the `false` and `null` cases as if `projectBaseUrl` was set. As we discussed, we're tempted to additionally validate that it's a string, but we'll keep it simple, since there's a lack of input validation in general in this project right now. 

```js
const program = { projectBaseUrl: false }
console.log(program.projectBaseUrl !== undefined); // true (gross)
```

```js
const program = { projectBaseUrl: null}
console.log(program.projectBaseUrl !== undefined); // true (gross)
```